internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -280,13 +280,13 @@ func (r *Reconciler) getNodeFromMachine(ctx context.Context, clusterClient clien
 
 // healthCheckTargets health checks a slice of targets
 // and gives a data to measure the average health.
-func (r *Reconciler) healthCheckTargets(targets []healthCheckTarget, logger logr.Logger, timeoutForMachineToHaveNode metav1.Duration) ([]healthCheckTarget, []healthCheckTarget, []time.Duration) {
+func (r *Reconciler) healthCheckTargets(targets []healthCheckTarget, l logr.Logger, timeoutForMachineToHaveNode metav1.Duration) ([]healthCheckTarget, []healthCheckTarget, []time.Duration) {
 	var nextCheckTimes []time.Duration
 	var unhealthy []healthCheckTarget
 	var healthy []healthCheckTarget
 
 	for _, t := range targets {
-		logger = logger.WithValues("Target", t.string())
+		logger := l.WithValues("Target", t.string())
```

Comment:
Would it be better to not pass in the logger to this function to make the new logger on each loop more explicit?

Reply:
What do you mean with "not pass in"?

Reply:
That would drop data of the other variables attached to the logger values. We could grab the logger from the context, but that would also require passing the context into this instead of the logger.